src/components/TokenImg.tsx

```diff
@@ -3,26 +3,12 @@ import styled from 'styled-components'
 
 import unknownTokenImg from 'img/unknown-token.png'
 
-const Wrapper = styled.img`
-  width: 30px;
-  height: 30px;
-`
-
-interface Props {
-  image: string
-  name: string
-  // required to be able to apply custom styles
-  // see https://www.styled-components.com/docs/basics#styling-any-component
-  className?: string
-}
-
 function _loadFallbackTokenImage(event: React.SyntheticEvent<HTMLImageElement>): void {
   const image = event.currentTarget
   image.src = unknownTokenImg
 }
 
-const TokenImg: React.FC<Props> = ({ image, name, className }: Props) => {
-  return <Wrapper src={image} alt={name} className={className} onError={_loadFallbackTokenImage} />
-}
-
-export default TokenImg
+export default styled.img.attrs(() => ({ onError: _loadFallbackTokenImage }))`
```

Comment:
If you opt not to depend on props in `attrs`, can just do an object instead of func
```
styled.img.attrs({ onError: _loadFallbackTokenImage })`...`
```

Reply:
Well, that should work as stated in the docs https://www.styled-components.com/docs/api#attrs, but for some reason it doesn't.

When I use the object instead of function, the `image` instance is `undefined` on the fallback function:

```
    TypeError: Cannot set property 'src' of undefined

       6 | function _loadFallbackTokenImage(event: React.SyntheticEvent<HTMLImageElement>): void {
       7 |   const image = event.currentTarget
    >  8 |   image.src = unknownTokenImg
         |            ^
       9 | }
      10 | 
      11 | export default styled.img.attrs({ onError: _loadFallbackTokenImage })`
```
Same for unit tests and the browser

Reply:
I experimented with this a little bit and found out
if a field of the attrs object is a function, then that property becomes a computed property, like
```
styled.img.attrs({ onError: (props) => _loadFallbackTokenImage })
// onError = _loadFallbackTokenImage

and with just
styled.img.attrs({ onError: _loadFallbackTokenImage })
// onError = _loadFallbackTokenImage(props)
```

Totally not obvious from the docs.

Reply:
Thanks for digging into this.
In that case, I'll stick to the current version.